### PR TITLE
feat: without-evm cli option in reth

### LINKED ISSUE #12090: Make setup_op_mainnet_without_ovm general purpose
### Describe the feature

This fills in dummy blocks:
https://github.com/paradigmxyz/reth/blob/fa59bd512e877e4677890b5c154b80816c07480e/crates/optimism/cli/src/commands/init_state/bedrock.rs#L12-L20

which can be useful for other chains as well https://github.com/paradigmxyz/reth/discussions/12057

currently this has hardcoded bedrock constants

## TODO
* refactor setup_op_mainnet_without_ovm so that it's general purpose

basically all of these constants should be an argument instead, or we transform all of this into a helper type instead
https://github.com/paradigmxyz/reth/blob/fa59bd512e877e4677890b5c154b80816c07480e/crates/optimism/cli/src/commands/init_state/bedrock.rs#L43-L43

cc @debjit-bw

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by debjit-bw**
Fixes #12090

Added CLI command to `init-state` without previously having to import blocks/receipts.

In order to try it out, you can use the following:
* [Example value for --header](https://github.com/debjit-bw/reth/blob/6cf525abb072cc20f95f29ef065412de6e6b55ea/header.rlp)
* [Example .jsonl state](https://github.com/debjit-bw/reth/blob/6cf525abb072cc20f95f29ef065412de6e6b55ea/test_dump.jsonl)

The optimism-specific implementation for op-geth hasn't been removed/modified.

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
hmm, this does now add the an additional command that isn't very useful for the regular ethereum CLI.

my idea for this was more like converting the existing pre-bedrock functions so that they are reusable and then move them as additional functions into the commands crate and optionally also include them into the init state command.

but this left the pre-bedrock functions untouched so far

**Comment by debjit-bw** [comment]
A unit test failed with this message:

```
The hosted runner: RethRunner_4145f40b940a lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.
```

**Review by joshieDo — APPROVED** [review]
some nits, but lgtm

mind adding the example as an actual test of `setup_without_evm` ?

File: crates/optimism/cli/src/commands/init_state.rs [hunk 1]
```diff
@@ -64,7 +68,7 @@ impl<C: ChainSpecParser<ChainSpec = OpChainSpec>> InitStateCommandOp<C> {
             } else if last_block_number > 0 && last_block_number < BEDROCK_HEADER.number {
                 return Err(eyre::eyre!(
                     "Data directory should be empty when calling init-state with --without-ovm."
-                ))
+                ));
```
[the remark this answers is not in the document]
Comment (debjit-bw) [reply crates/optimism/cli/src/commands/init_state.rs]: lol mp, just pushed the fix

**Comment by debjit-bw** [comment]
@mattsse can you have a look?

File: crates/cli/commands/src/init_state/mod.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    /// Specifies whether to initialize the state without relying on EVM historical data.
+    ///
+    /// When enabled, and before inserting the state, it creates a dummy chain up to the last EVM
+    /// block specified. It then, appends the first block provided block.
+    ///
+    /// - **Note**: **Do not** import receipts and blocks beforehand, or this will fail or be
+    ///   ignored.
+    #[arg(long, default_value = "false")]
+    without_evm: bool,
```
Comment (mattsse) [inline crates/cli/commands/src/init_state/mod.rs:54]:
```suggestion
    pub without_evm: bool,
```

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**